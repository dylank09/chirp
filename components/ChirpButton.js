import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import { theme } from "../assets/Theme";

export default function ChirpButton(props) {
  return (
    <TouchableOpacity
      onPress={props.onPress}
      style={[styles.button, { width: props.width }]}
    >
      <Text style={styles.buttonText}>{props.text ? props.text : "  "} </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: theme.colors.button,
    padding: 10,
    margin: 10,
    borderRadius: 18,
    alignItems: "center",
    height: theme.dimensions.buttonHeight,
  },
  buttonText: {
    color: theme.colors.buttonText,
    fontSize: theme.dimensions.standardFontSize,
  },
});
