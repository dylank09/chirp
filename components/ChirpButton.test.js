import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

import ChirpButton from "./ChirpButton";

describe("<ChirpButton />", () => {
  const tree = renderer.create(<ChirpButton />).toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("button is present", () => {
    const { getByTestId } = render(
      <ChirpButton
        onPress={() => console.log("pressed")}
        width="50%"
        text="Testing Button"
      />
    );
    const foundButton = getByTestId("button");
    expect(foundButton).toBeTruthy();
  });
});
