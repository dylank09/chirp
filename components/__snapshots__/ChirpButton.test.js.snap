// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChirpButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(193, 73, 83)",
      "borderRadius": 18,
      "height": undefined,
      "margin": 10,
      "opacity": 1,
      "padding": 10,
      "width": undefined,
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
        "fontSize": 15,
      }
    }
  >
      
     
  </Text>
</View>
`;
